closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -47,6 +48,46 @@
 # include <pthread.h>
 #endif /* defined(WINDOWS) */
 
+// This definition is needed to compile the "OSSL_RSA_get0_key_compatibility" function
+// Since the first parameter of that function is of type RSA, which is
+// a typedef of rsa_st, the function would not compile without defining
+// rsa_st here
+struct rsa_st {
```

Comment:
This is not a good idea; `rsa_st` is intended to be an opaque type - see https://github.com/openssl/openssl/issues/1604#issuecomment-248237417.

Reply:
From the mentioned comment above, my understanding is that this opacity principle is added in OpenSSL 1.1.x. The structure I add here is for the JDK's compatibility with OpenSSL 1.0.x.

More specifically, the `rsa_st` defined here is needed to compile `OSSL_RSA_get0_key_compatibility`. The last is only used when the JDK cannot find the function `RSA_get0_key` ([corresponding source code](https://github.com/ibmruntimes/openj9-openjdk-jdk11/blob/8c626311828f8601f0be3165ee136d4393b0b82c/closed/src/java.base/share/native/libjncrypto/NativeCrypto.c#L427-L429)). That function is found only in OpenSSL 1.1.x and not in OpenSSL 1.0.x. Hence, the only time we use `OSSL_RSA_get0_key_compatibility` is in OpenSSL 1.0.x.

So the opacity principle of OpenSSL 1.1.x is respected, but this definition here is needed to compile a function for backward compatibility.